src/nspawn/nspawn.c

```diff
@@ -1402,6 +1403,25 @@ static int setup_journal(const char *directory) {
         id = strstrip(b);
         if (isempty(id) && try)
                 return 0;
+        else if (isempty(id)) {
+                sd_id128_t new_id;
+                char s[33] = {};
+
+                /* Machine ID is empty let's populate new one for this container */
+                r = sd_id128_randomize(&new_id);
+                if (r < 0)
+                        return log_error_errno(r, "Faile to generate random machine ID: %m");
+
+                r = machine_id_setup(directory ?: "/", new_id);
```

Comment:
machine_id_setup() is happy to take NULL, no? No need to convert this to "/" in case of NULL...